### PR TITLE
gotags: reinit at 1.4.1

### PR DESCRIPTION
**Pull request by urandom2**
###### Description of changes

This change reverts #176044 and #178000, as well as updating the version
to the latest stable release. Unfortunately, despite the lack of updates
to gotags, it is still depended upon by vim-go, thus we must keep it
around until they (and other consumers) have removed it from the
dependency trees.

This change also adds myself as a maintainer, since I would not wish the
burden of this package's maintanence on anybody else.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [r] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by bjornfor — APPROVED** [review]

**Comment by bjornfor** [comment]
Nit: it's preferred/cleaner to add yourself to the central maintainer list in a separate commit, and then start referencing that entry in subsequent commits.

**Comment by arnottcr** [comment]
so, two commits in this pr, or two prs? (happy to make the change)

**Comment by bjornfor** [comment]
Whatever you find easiest.

**Comment by arnottcr** [comment]
I went for different commit same pr, so as to reduce the maintenance burden on the rest of the core team; happy to oblige, hope this helps!

**Review by bjornfor — APPROVED** [review]

File: pkgs/development/tools/gotags/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  src = fetchFromGitHub {
+    owner = "jstemmer";
+    repo = pname;
+    rev = "4c0c4330071a994fbdfdff68f412d768fbcca313";
+    sha256 = "sha256-cHTgt+zW6S6NDWBE6NxSXNPdn84CLD8WmqBe+uXN8sA=";
+  };
+
+  meta = with lib; {
+    description = "ctags-compatible tag generator for Go";
+    homepage = "https://${goPackagePath}";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/gotags/default.nix:18]:
```suggestion
  src = fetchFromGitHub {
    owner = "jstemmer";
    repo = pname;
    rev = "4c0c4330071a994fbdfdff68f412d768fbcca313";
    sha256 = "sha256-cHTgt+zW6S6NDWBE6NxSXNPdn84CLD8WmqBe+uXN8sA=";
  };

  goPackagePath = "github.com/jstemmer/gotags";

  meta = with lib; {
    description = "ctags-compatible tag generator for Go";
    homepage = "https://github.com/jstemmer/gotags";
```

Comment (arnottcr) [reply pkgs/development/tools/gotags/default.nix:18]: happy to move `goPackagePath` down, but is it critical that we repeat ourselves?

Comment (SuperSandro2000) [reply pkgs/development/tools/gotags/default.nix:18]: Yes, meta.homepage should be clickable to make reviewer lives easier.

Comment (arnottcr) [reply pkgs/development/tools/gotags/default.nix]: sorry for the delay, I rebased and everything should be done

Comment (SuperSandro2000) [reply pkgs/development/tools/gotags/default.nix]: No, please keep it simple. The homepage could also be something else like the docs or website.

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -13498,6 +13498,15 @@
     githubId = 619015;
     name = "Svintsov Dmitry";
   };
+  urandom = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:13501]:
```suggestion
  arnottcr = {
```
Can you please use a name that is somehow related to your existing profile to make it easier to request review from or ping you?

File: pkgs/development/tools/gotags/default.nix (hunk 1 above)
Comment (arnottcr) [reply pkgs/development/tools/gotags/default.nix]: sorry, looks like it is just the repo; I am mostly mirroring the old derivation, and linking to pkg.go.dev seems like bad form.

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (arnottcr) [reply maintainers/maintainer-list.nix:13501]: my online identity is currently centered around the `urandom` handle, unfortunately my github account dates back to high school, and somebody else took that handle before I could correct things here.

you will also note my email domain is `urandom.co.uk`, and I am urandom on [discourse](https://discourse.nixos.org/u/urandom), [sourcehut](https://git.sr.ht/~urandom), and other platforms.

**Pull request merged**